ddtrace/internal/forksafe.py

```diff
@@ -0,0 +1,101 @@
+"""
+An API to provide after_in_child fork hooks across all Pythons.
+"""
+import functools
+import logging
+import os
+import threading
+
+
+__all__ = [
+    "call_nocheck",
+    "ddtrace_after_in_child",
+    "register",
+]
+
+
+log = logging.getLogger(__name__)
+
+BUILTIN_FORK_HOOKS = hasattr(os, "register_at_fork")
+
+registry = []
+
+
+def ddtrace_after_in_child():
+    global registry
+
+    for hook in registry:
+        try:
+            hook()
+        except Exception:
+            # Mimic the behaviour of Python's fork hooks.
+            log.exception("Exception ignored in forksafe hook %r", hook)
+
+
+def _register(hook):
+    if hook not in registry:
+        registry.append(hook)
+
+
+if BUILTIN_FORK_HOOKS:
+
+    def register(after_in_child):
+        _register(after_in_child)
+        return lambda f: f
```

Comment:
hm don't you need to handle `_check_pid` here? just removing it from the `kwargs`?

Reply:
No because it won't even be included. That's why `call_nocheck` exists.

If we had the alternate approach of having the user explicitly include `_check_pid` instead of calling `call_nocheck` then we'd need to pop it here.

Reply:
I understood the user had that possibility when reading the case where we have no builtin fork hooks 🤔

Reply:
I mean they could in theory pass the argument manually, but that's why the function exists, to cover that implementation detail so that the user doesn't have to worry about that. Else we'd be needlessly passing that argument in this case where fork hooks are available.